[BE] Add dtypesIfMPS to ModuleInfo enabling float16 tests for MPS and remove all skipIfMPS for float64

mikaylagawarecki opened the pull request:
Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119039

mikaylagawarecki committed:
Add dtypesIfMPS to ModuleInfo and remove all float64 skips

[ghstack-poisoned]

janeyx99 commented on torch/testing/_internal/common_modules.py:
```diff
@@ -211,6 +211,7 @@ def __init__(self,
                  skips=(),  # Indicates which tests to skip
                  decorators=None,  # Additional decorators to apply to generated tests
                  dtypes=floating_types(),  # dtypes this function is expected to work with
+                 dtypesIfMPS=(torch.float32,),  # dtypes this function is expected to work with on MPS
```
Is this necessary if it's always torch.float32?

mikaylagawarecki replied:
Hm I was thinking that if MPS adds support for other dtypes then this kwarg would be useful

malfet replied:
MPS supports float16, so please add it here

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo and remove all skipIfMPS for float64"

Right now, ModuleInfo.dtypes defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

janeyx99 commented on torch/testing/_internal/common_modules.py:
```diff
@@ -211,6 +211,7 @@ def __init__(self,
                  skips=(),  # Indicates which tests to skip
                  decorators=None,  # Additional decorators to apply to generated tests
                  dtypes=floating_types(),  # dtypes this function is expected to work with
+                 dtypesIfMPS=(torch.float16, torch.float32,),  # dtypes this function is expected to work with on MPS
```
I am curious how we would scale this in the future when there may be more accelerators. I'd guess we'd have an API somewhat like `accelerator.get_supported_dtypes()` and the `dtypes` field here could be a dynamic function that gets the current accelerator and returns the relevant dtypes, similar to your supported_dtypes function.

This then brings me to think "hey, maybe in the future there would be an accelerator-aware `supported_floating_types()` vs what's used in line 213". Anyway, this discussion is likely outside the scope of this PR. cc'ing @albanD in case he has thoughts

mikaylagawarecki replied:
Great point! I was thinking about this too :)

I don't think we will have BC issues changing the dtype logic in the test infra. Updating `dtypes` to account for the current accelerator in the future definitely sounds reasonable to me

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo enabling float16 tests and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo enabling float16 tests for MPS and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo enabling float16 tests for MPS and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot merge

No MPS failures on PR

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo enabling float16 tests for MPS and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki committed:
Update on "[BE] Add dtypesIfMPS to ModuleInfo enabling float16 tests for MPS and remove all skipIfMPS for float64"

Right now, `ModuleInfo.dtypes` defaults to `torch.testing._internal.common_dtype.floating_types()`, almost no ModuleInfos override this (so only `float32` and `float64` are tested).

This is the first step to clean up/improve dtype testing for `ModuleInfos` and fix #116626.

Follow up PRs will updates `dtypes=` (and perhaps `dtypesIf{Device}` (if it makes sense)) for each `ModuleInfo`




[ghstack-poisoned]

mikaylagawarecki commented:
@pytorchbot rebase -s

mikaylagawarecki commented:
@pytorchbot merge
